In [2]:
import pandas as pd
import numpy as np

## ANOVA DE UNA VÍA

In [3]:
df_datosbaseball=pd.read_excel('data/datosbaseball.xlsx')
df_datosbaseball.head()

posicion  bateo
0       OF  0.359
1       IF  0.340
2       IF  0.330
3       OF  0.341
4       IF  0.366

In [4]:
import statsmodels.api as sm
from statsmodels.formula.api import ols

In [5]:
model=ols('bateo ~ posicion',data=df_datosbaseball).fit()

In [6]:
anova_table=sm.stats.anova_lm(model,typ=2)
anova_table

sum_sq     df         F    PR(>F)
posicion  0.007557    3.0  1.994349  0.114693
Residual  0.407984  323.0       NaN       NaN

pvalue=0.114693 es mayor que alfa así que no podemos rechazar la hipótesis nula de que las medias son iguales y por tanto no podemos afirmar que la posición tenga una influencia en el bateo

## ANOVA DE BLOQUES O MEDIAS REPETIDAS

In [8]:
df_datostienda=pd.read_excel('data/datostienda.xlsx')
df_datostienda.head()

producto tienda  precio
0   lettuce      A   1.755
1  potatoes      A   2.655
2      milk      A   2.235
3      eggs      A   0.975
4     bread      A   2.370

In [9]:
model=ols('precio ~ tienda + producto',data=df_datostienda).fit()

In [10]:
anova_table=sm.stats.anova_lm(model,typ=2)
anova_table

sum_sq    df           F        PR(>F)
tienda      5.737207   3.0   13.025212  1.897557e-05
producto  139.479418   9.0  105.553625  1.102247e-18
Residual    3.964224  27.0         NaN           NaN

Utilizamos el estádictico f de nuestro estudio F=13.025212, el pvalue=1.897557e-05 es menos que alfa, rechazamos la hipótesis nula de que todas las medias son iguales y por tanto la tienda tiene un efecto en el precio.

En el análisis por bloques tenemos el efecto que tiene el precio de la compra F=105.553625 (no nos interesa) el pvalue=1.102247e-18 nos dice que los bloques tienen un efecto en el precio y hemos hecho bien en meter el bloque ya que hemos reducido el error al meter la influencia del producto

Por lo tanto, vemos como tanto la precia como el producto tienen una influencia en la media de precios.

## ANOVA DE DOS VÍAS

In [61]:
datosconstruct=pd.read_excel('data/datosconstruct.xlsx')
datosconstruct.head()

templado  grosor  resistencia
0   rapido       8        15.29
1   rapido       8        15.89
2   rapido       8        16.02
3   rapido       8        16.56
4   rapido       8        15.46

In [63]:
datosconstruct['grosor']=datosconstruct['grosor'].astype('category')
datosconstruct.head()

templado grosor  resistencia
0   rapido      8        15.29
1   rapido      8        15.89
2   rapido      8        16.02
3   rapido      8        16.56
4   rapido      8        15.46

In [57]:
model=ols('resistencia ~ templado * grosor',data=datosconstruct).fit()
anova_table=sm.stats.anova_lm(model,typ=2)
anova_table

sum_sq    df           F        PR(>F)
templado         112.675320   1.0  380.082037  3.194915e-16
grosor            10.413167   2.0   17.563108  2.000663e-05
templado:grosor    1.603500   2.0    2.704503  8.725105e-02
Residual           7.114800  24.0         NaN           NaN

Ho 1 = valor medio resistencia (templado lento) = valor medio resistencia (templado rapido)  --> Por lo tanto, la Hipotesis nula es que el tipo de templado no afetca en la resistencia.

Con estos Pvalues tan bajos, podemos rechazar casi con total seguridad la hipotesis nula. Por lo tanto, tanto el templado como el grosor tienen una influencia en la resistencia. 
Además, vemos tambien que entre ellos no estan relacionados (Pvalue templado:grosor > 0.05).